exercises/dominoes/src/test/java/DominoesTest.java

```diff
@@ -0,0 +1,189 @@
+		
+		assertNull(chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void disconnectedSimpleTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 1), new Domino(2, 2)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertNull(chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void disconnectedDoubleLoopTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 1), new Domino(3, 4), new Domino(4, 3)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertNull(chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void disconnectedSingleIsolatedTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(4, 4)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertNull(chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void needBacktrackTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(2, 4), new Domino(4, 2)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void separateLoopsTest() {
+		Dominoes dominoes = new Dominoes();
+		
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(2, 3), new Domino(3, 1), new Domino(1, 1), new Domino(2, 2), new Domino(3, 3)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	@Ignore("Remove to run test")
+	@Test
+	public void nineElementsTest() {
+		Dominoes dominoes = new Dominoes();
+		Domino[] dominoesList = {new Domino(1, 2), new Domino(5, 3), new Domino(3, 1), new Domino(1, 2), new Domino(2, 4), new Domino(1, 6), new Domino(2, 3), new Domino(3, 4), new Domino(5, 6)};
+		ArrayList<Domino> dominoesArray = new ArrayList<>(Arrays.asList(dominoesList));
+		
+		ArrayList<Domino> chain = dominoes.formChain(dominoesArray);
+		
+		assertValidChain(dominoesArray, chain);
+	}
+	
+	private void assertValidChain(ArrayList<Domino> inputDominoes, ArrayList<Domino> outputDominoes) {
+		assertSameDominoes(inputDominoes, outputDominoes);
+		assertEndDominoesMatch(outputDominoes);
+		assertConsecutiveDominoes(outputDominoes);
+	}
+	
+	private void assertEndDominoesMatch(ArrayList<Domino> outputDominoes) {
```

Comment:
For this and the other two helper methods it might be useful to provide a useful error message to assertEquals? Otherwise the user will just get a message along the lines of `Expected 3 but got 4`. Which isn't very useful. The error message should contain all the details the user needs to see exactly what's gone wrong 🙂